File: Selenium/Project/Activity5.java
package project;
import static org.testng.Assert.assertEquals;

import java.time.Duration;
import java.util.concurrent.TimeUnit;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import net.bytebuddy.description.modifier.Visibility;



public class Activity5 
{
	WebDriver driver ;
	WebDriverWait wait;
	 @BeforeClass
	    public void beforeClass() {
	        driver = new FirefoxDriver();
	        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
	        driver.get("http://alchemy.hguy.co:8080/orangehrm/symfony/web/index.php/auth/login");
	    }
	
	 @Test
	    public void test() throws InterruptedException {
		
 	    driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(6));

	        WebElement username = driver.findElement(By.xpath("//input[contains(@name,'Username')]"));
	        WebElement password = driver.findElement(By.xpath("//input[contains(@name,'Password')]"));
	        WebElement login = driver.findElement(By.xpath("//input[@name='Submit']"));
	        
	        username.sendKeys("orange");
	        password.sendKeys("orangepassword123");
	        login.click();
	        Thread.sleep(10);
	        
	        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
	        
	        WebElement my_info = driver.findElement(By.xpath("//a[@id='menu_pim_viewMyDetails']"));
	        my_info.click();
	        WebElement edit = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//input[@value='Edit']")));
	        edit.click();

	        WebElement firstName = driver.findElement(By.xpath("//input[@id='personal_txtEmpFirstName']"));
	        WebElement lastName = driver.findElement(By.xpath("//input[@id='personal_txtEmpLastName']"));
	        WebElement gender = driver.findElement(By.xpath("//input[@id='personal_optGender_2']"));
	        WebElement nationality_dropdown = driver.findElement(By.xpath("//select[@id='personal_cmbNation']"));
	        Select select = new Select(nationality_dropdown);
	        
	        firstName.clear();
	        firstName.sendKeys("Mary");
	        lastName.clear();
	        lastName.sendKeys("abhraham");
	        gender.click();
	        select.selectByVisibleText("Indian");

	        driver.findElement(By.xpath("//input[@id='btnSave']")).click();
	    	
	    }

	    @AfterClass
	    public void tearDown() {
	        driver.quit();
	    }


}

File: Selenium/Project/Activity5.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

with webdriver.Firefox() as driver:
    driver.get("http://alchemy.hguy.co:8080/orangehrm/symfony/web/index.php/auth/login")
    username_field = driver.find_element(By.XPATH,"//input[contains(@name,'Username')]")
    password_field = driver.find_element(By.XPATH,"//input[contains(@name,'Password')]")
    login_button = driver.find_element(By.XPATH,"//input[@name='Submit']")
    
    username_field.send_keys("orange")
    password_field.send_keys("orangepassword123")
    login_button.click()

    wait = WebDriverWait(driver, timeout=10, poll_frequency=3, ignored_exceptions='NoSuchElementException' )
    
    my_info = wait.until(EC.visibility_of_element_located((By.XPATH,"//a[@id='menu_pim_viewMyDetails']")))
    my_info.click()
    edit = wait.until(EC.visibility_of_element_located((By.XPATH,"//input[@value='Edit']")))
    edit.click()

    firstName_field = driver.find_element(By.XPATH,"//input[@id='personal_txtEmpFirstName']")
    lastName_field = driver.find_element(By.XPATH,"//input[@id='personal_txtEmpLastName']")
    gender = driver.find_element(By.XPATH,"//input[@id='personal_optGender_2']")
    nationality_dropdown = driver.find_element(By.XPATH,"//select[@id='personal_cmbNation']")
    select = Select(nationality_dropdown) 
    
    firstName_field.clear()
    firstName_field.send_keys("Mary")
    lastName_field.clear()
    lastName_field.send_keys("John")
    gender.click()
    select.select_by_visible_text("Indian")

    driver.find_element(By.XPATH,"//input[@id='btnSave']").click()
    driver.quit()